### PR TITLE
feat: add option to specify additional certs

### PR DESCRIPTION
**Pull request by sreya**
This is the first part of the delivery for supporting extra certs. A lot of the code here is test code to facilitate a good integration test. This PR allows us to instantiate a client that we use to push build logs to the Coder control plane.

The next piece will deliver support for copying the certs into the correct directory when dialing a private Docker registry.

### CODE REVIEW

File: integration/integrationtest/coder.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			Certificates: []tls.Certificate{cert},
+		}
+		s.StartTLS()
+	} else {
+		s.Start()
+	}
+
+	t.Cleanup(s.Close)
+	return recorder
+}
```
Comment (johnstcn) [inline integration/integrationtest/coder.go:95]: This endpoint is deprecated. Would it make sense to instead test the DRPC agent endpoint for submitting logs? 
I was able to partially test it by faking out the DRPC server e.g. https://github.com/coder/envbuilder/blob/main/log/coder_internal_test.go#L119-L141

Comment (sreya) [reply integration/integrationtest/coder.go:95]: It's deprecated but we're really just testing that the client can establish a connection over TLS with the provided cert pool. The actual logic of sending build logs is less important imo.

File: xhttp/client.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		return &http.Client{}, nil
+	}
+
+	pool, err := certPool(log, extraCertsPath)
+	if err != nil {
+		return nil, xerrors.Errorf("cert pool: %w", err)
+	}
+
+	//nolint:forcetypeassert
+	transport := (http.DefaultTransport.(*http.Transport)).Clone()
```
[the remark this answers is not in the document]
Comment (sreya) [reply xhttp/client.go:27]: It's not clear to me what this offers over simply cloning the transport? It looks like it was created prior to the `Clone` function being added.

[the remark this answers is not in the document]
Comment (johnstcn) [reply xhttp/client.go:27]: Fair 👍

**Review by johnstcn — APPROVED** [review]
I don't have any further blocking comments 👍

**Pull request merged**